client/client.go

```diff
@@ -210,13 +216,17 @@ func (c *Client) update(latestRoot bool) (data.TargetFiles, error) {
 
 	// If we don't have the root.json, download it, save it in local
 	// storage and restart the update
-	if !c.hasMetaFromSnapshot("root.json", rootMeta) {
-		return c.updateWithLatestRoot(&rootMeta)
+	// Root should no longer be pinned in snapshot meta https://github.com/theupdateframework/tuf/pull/988
+	if rootMeta, ok := snapshotMetas["root.json"]; ok {
+		if !c.hasMetaFromSnapshot("root.json", rootMeta) {
+			return c.updateWithLatestRoot(&rootMeta)
+		}
 	}
 
 	// If we don't have the targets.json, download it, determine updated
 	// targets and save targets.json in local storage
 	var updatedTargets data.TargetFiles
+	targetsMeta := snapshotMetas["targets.json"]
```

Comment:
since this is a client downloading possibly bad metadata, does this always have to exist? should it fail early?

It seems like it was always assumed to exist in the previous code too

Reply:
maybe i'm confused what's being downloaded unsafely or not...

Reply:
on this part I kept the existing behavior, would you rather have me change it and skip the targets step if no target are found in the validated snapshot?
If no targets are set in the snapshot meta, the current logic will try to download a targets of version 0. It will fail if previous targets were seen on the version check.

Reply:
I *think* the existing behavior should be fine based on the tests, so I would say that's good!